common4j/src/main/com/microsoft/identity/common/java/crypto/StorageEncryptionManager.java

```diff
@@ -482,7 +482,8 @@ private void assertHMac(final byte[] encryptedBlob,
         }
 
         if (result != 0) {
-            throw new ClientException(HMAC_MISMATCH);
+            final ClientException ex = new ClientException(HMAC_MISMATCH);
+            throw ExceptionAdapter.trimStackTrace(ex, 4);
```

Comment:
> throw ExceptionAdapter.trimStackTrace(ex, 4); [](http://example.com/codeflow?start=11&length=46)

it will almost always be same 4 lines.

The reason there are frequent assertHMac logs because storageEncryptionManger's getData() calls this and logs the stack. 

I was thinking we can remove

            Logger.error(
                TAG + methodName,
                "Decryption failure. "
                        + WorkplaceJoinFailure.INTERNAL, e
            )

instead
            Logger.error(
                TAG + methodName,
                "Decryption failure. "
                        + WorkplaceJoinFailure.INTERNAL + e.getCode() + e.getMessage(), null
            )

and log "key" as well. in AbstractEncryptedBrokerAccountStorage